qiskit/algorithms/gradients/qfi.py

```diff
@@ -0,0 +1,81 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022, 2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+A class for the Quantum Fisher Information.
+"""
+
+from __future__ import annotations
+
+from abc import ABC
+from collections.abc import Sequence
+from typing import Literal
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.primitives import BaseEstimator
+from qiskit.providers import Options
+
+from .lin_comb_estimator_gradient import DerivativeType
+from .lin_comb_qgt import LinCombQGT
+from .qgt_result import QGTResult
+
+METHOD = Literal["lin_comb"]
+
+
+class QFI(ABC):
+    r"""Computes the Quantum Fisher Information (QFI) given a pure,
+    parameterized quantum state. QFI is defined as:
+
+    .. math::
+
+        \mathrm{QFI}_{ij}= 4 \mathrm{Re}[\langle \partial_i \psi | \partial_j \psi \rangle
+        - \langle\partial_i \psi | \psi \rangle \langle\psi | \partial_j \psi \rangle].
+    """
+
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        phase_fix: bool = True,
+        method: METHOD = "lin_comb",
+        options: Options | None = None,
```

Comment:
I think the cleaner solution here would be to just allow a `BaseQGT` as input, otherwise adding a new `QGT` method would mean we also have to change this class here. We could also add alternate constructors but maybe it would be best to do this along with the other gradients 🙂

Reply:
I was thinking to make something like this that the opflow gradients have. Yes, I was thinking to modify `QFI` each time a new `QGT` is added. Then we can have an entry point that users can easily select a QFI method by typing method's  name.


https://github.com/Qiskit/qiskit-terra/blob/3d8a6e03931c26b1132b3f4cea4015bbff998d63/qiskit/opflow/gradients/qfi_base.py#L43-L48

Reply:
It would be nice to have a convenient construction but maybe we should reconsider how these work as we have to implement those for all gradients. E.g. do we want to use strings or factories (which can have additional attributes like finite difference perturbations)? Maybe we could even just specify the technique instead of the full gradient class, e.g.
```python
algo = MyAlgorithm(gradient=ParamShift(), qgt=LinComb())
```
But since that's not clear yet maybe it would be better to hold this string construction off for now 🤔 With the QGT implemented we can also use that one as "primary" object to use in VarQTE so maybe the QFI is less used/important now 🙂

Reply:
Ah, it seems we have a lot of things to discuss (such as additional attributes like finite difference perturbations and so on)🤔. Specifying the technique in  algorithm classes look nice too. 
I will change to allow a BaseQGT as input.